doc/architecture/decisions/003-authoring-schema-variants.md

```diff
@@ -0,0 +1,41 @@
+Schema variations are tackled through a mixture of routing rules (go here if...), skip conditions (skip this if...) and splitting of schemas into "form types" (e.g. RSI form types 0102, 0112 are survey runner schemas 1_0102.json and 1_0112.json). Compared to existing surveys MBS has a large number of form types which makes splitting and maintaining them by hand too onerous. The Author service is the long term solution but we need something more manageable in the interim to make management of MBS content more cost effective.
+
+We need a solution that lightens the load of authoring form variations.
+
+## Decision
+
+We will have a single data source per survey and update the build process so that the survey sources are turned into multiple survey runner schema compliant json files.
+
+Each survey data source will be made up of one or more manifest files and one or more block files. Each manifest file will represent a form type. A manifest will define the survey metadata, one or more groups and the block ids that each group composes. Manifests will be stored in YAML.
+
+A block file will define a single survey runner block and the question it contains. Blocks which have different metadata but contain the same question will duplicate the question. Blocks which contain the same question should have the same file name with a prefix that makes identifying potential duplicate/similar questions easy, e.g. for RSI `0102-total-internet-sales`, `0112-total-internet-sales`. Blocks will be stored in YAML.
+
+Manifests will live in `data/source/manifests/[survey]/` e.g. for RSI `data/source/manifests/rsi/0102.manifest.yaml
+
+Blocks will live in `data/source/blocks/[survey]/` e.g. for RSI `data/source/blocks/rsi/internet-sales.block.yaml`
+
+The scripts which aid the build process and other useful tasks will live in the `scripts/` folder.
+
+The json files produced by the manifests will go through a schema validation process. If one of the json schemas fails validation the entire build will fail.
+
+This solution is temporary and will be superseded once Author goes live.
+
+## Consequences
+
+Authors will switch from creating a whole schema for each form type to creating a manifest for each form type and any blocks they require. This means there will be a high up front cost for authoring the first form type with greater returns for any future authoring, depending on how many blocks can be re-used.
+
+Questions do not have a single source of truth. A question can appear in many blocks, where typically the blocks have different routing and/or skip conditions. As questions are embedded in block files they are effectively duplicated. Authors who are updating a question, e.g. a guidance change, will have to bear this in mind. They will have to identify the blocks that contain the question and update each of them.
+
+Authors will need to remember to run the build before they submit a pull request to make sure any changes they have made don't violate Runners schema.
```

Comment:
Related to above, as long as the survey runner JSON files are generated before the tests are run, the existing unit test will pick these things up.